tests/CsvParserTest.py

```diff
@@ -0,0 +1,63 @@
+        input_file = "tests/test-files/csv/extended.csv"
+        result = CsvParser().parse_file(input_file, self.options)
+        expected_library_mapping = {
+            pathlib.Path('foo/bar.vhd').absolute().resolve(): 'test1',
+            pathlib.Path('clock_generator.vhd').absolute().resolve(): 'test2',
+            pathlib.Path('dut.vhd').absolute().resolve(): 'test2',
+            pathlib.Path('testbench.vhd').absolute().resolve(): 'test1'
+        }
+        expected_includes = {
+            pathlib.Path('includes').absolute().resolve()
+        }
+        expected_defines = ['SIGASI=true']
+        self.assertDictEqual(result.library_mapping, expected_library_mapping, 'Library mapping mismatch')
+        self.assertSetEqual(result.verilog_includes, expected_includes, 'Includes list mismatch')
+        self.assertListEqual(result.verilog_defines, expected_defines, 'Defines list mismatch')
+
+    def test_multimap_csv(self):
+        self.maxDiff = None
+        command_line_options = ['the_project', 'tests/test-files/csv/multimap.csv']
+        self.options = ProjectOptions(command_line_options)
+        input_file = "tests/test-files/csv/multimap.csv"
+        result = CsvParser().parse_file(input_file, self.options)
+        expected_library_mapping = {
+            pathlib.Path('foo/bar.vhd').absolute().resolve(): 'test1',
+            pathlib.Path('clock_generator.vhd').absolute().resolve(): ['test2', 'test1', 'test7'],
+            pathlib.Path('dut.vhd').absolute().resolve(): ['test2', 'test3'],
+            pathlib.Path('testbench.vhd').absolute().resolve(): 'test1'
+        }
+        self.assertDictEqual(result.library_mapping, expected_library_mapping, 'Library mapping mismatch')
+        self.assertFalse(result.verilog_includes, 'Includes list mismatch')
+        self.assertFalse(result.verilog_defines, 'Defines list mismatch')
+
+    def test_not_extended_csv(self):
+        command_line_options = ['the_project', 'tests/test-files/csv/legacy.csv', '--no-extended-csv']
+        self.options = ProjectOptions(command_line_options)
+        input_file = "tests/test-files/csv/legacy.csv"
+        result = CsvParser().parse_file(input_file, self.options)
+        expected_library_mapping = {
+            pathlib.Path('foo/bar.vhd').absolute().resolve(): 'test1',
+            pathlib.Path('clock_generator.vhd').absolute().resolve(): 'test2',
+            pathlib.Path('bahr/define.sv').absolute().resolve(): 'define',
+            pathlib.Path('dut.vhd').absolute().resolve(): 'test2',
+            pathlib.Path('include.vhd').absolute().resolve(): 'include',
+            pathlib.Path('testbench.vhd').absolute().resolve(): 'test1'
+        }
+        self.assertDictEqual(result.library_mapping, expected_library_mapping, 'Library mapping mismatch')
+        self.assertFalse(result.verilog_includes, 'Includes list mismatch')
+        self.assertFalse(result.verilog_defines, 'Defines list mismatch')
```

Comment:
Rather than `assertFalse`, use `assertEquals(..., False` or actually `assertSetEqual(..., set()` and vice versa for the second.

Reply:
Wait no, these return None, not an empty collection. Use `assertIsNone`.

(If we need to return `None`, I'd prefer returning an empty collection actually.)

Reply:
Nvm they do return an empty set; please use `assertSetEqual` or even `assertEqual(len(...), 0, ...`